src/ir/lowering.rs

```diff
@@ -1396,6 +1495,36 @@ fn lower_fn_call(
                 })?
         }
     };
+
+    let mut generic_map = HashMap::new();
+
+    if let Some(generic_fn_def) = builder.ctx.generic_fn_bodies.get(&fn_id) {
+        // todo: fix this clone
+
+        if info.generics.len() != generic_fn_def.decl.generic_params.len() {
+            // todo: error mismatch generics
+            panic!();
```

Comment:
I understand that this panic is here because there is more left to implement but we should fail with a proper error message here